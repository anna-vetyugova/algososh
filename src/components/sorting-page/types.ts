export type TItem = {
  index: number;
  sortedArray: number[][];
  currentStepIndex: number;
};

