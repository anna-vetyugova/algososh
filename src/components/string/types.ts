export type TStringIntem = {
  index: number;
  steps: string[][];
  currentStepIndex: number;
};
