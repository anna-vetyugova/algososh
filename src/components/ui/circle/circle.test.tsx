import React from 'react';
import {render, screen} from '@testing-library/react'

import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

describe( 'Circle component', () => {
  it('renders without letter', () => {
    render(<Circle letter={''} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with letter', () => {
    render(<Circle letter={'letter'} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with head', () => {
    render(<Circle head={'head'} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with react-element-head', () => {
    render(<Circle head={<Circle />} />);
    const circleElement = screen.getAllByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with tail', () => {
    render(<Circle tail={'tail'} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with react-element-tail', () => {
    render(<Circle tail={<Circle />} />);
    const circleElement = screen.getAllByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with index', () => {
    render(<Circle index={1} />);
    const circleElement = screen.getAllByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with isSmall prop', () => {
    render(<Circle isSmall={true} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with Default state', () => {
    render(<Circle state = {ElementStates.Default} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with Changing state', () => {
    render(<Circle state = {ElementStates.Changing} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
  it('renders with Modified state', () => {
    render(<Circle state = {ElementStates.Modified} />);
    const circleElement = screen.getByTestId('circle');
    expect(circleElement).toMatchSnapshot();
  });
});