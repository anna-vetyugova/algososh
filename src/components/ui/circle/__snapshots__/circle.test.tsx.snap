// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component renders with Changing state 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders with Default state 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders with Modified state 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders with head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders with index 1`] = `
Array [
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>,
]
`;

exports[`Circle component renders with isSmall prop 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders with letter 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders with react-element-head 1`] = `
Array [
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>,
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>,
]
`;

exports[`Circle component renders with react-element-tail 1`] = `
Array [
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>,
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>,
]
`;

exports[`Circle component renders with tail 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle component renders without letter 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
